lib/Util/TextFormat.php

```diff
@@ -16,11 +16,27 @@ public function __construct(string $indentation = '    ')
 
     public function indent(string $string, int $level = 0)
     {
-        $lines = explode(PHP_EOL, $string);
+        $newLineChar = TextUtil::newLineChar($string);
+        $lines = TextUtil::lines($string);
         $lines = array_map(function ($line) use ($level) {
+            if (!$line) {
+                return $line;
+            }
             return str_repeat($this->indentation, $level) . $line;
         }, $lines);
 
-        return implode(PHP_EOL, $lines);
+        return implode($newLineChar, $lines);
+    }
+
+    public static function indentationRemove(string $text): string
+    {
+        $text = preg_replace("/^ +/m", "", $text);
```

Comment:
That won't remove tabulation, does it ?
We should handle all vertical spaces here don't you think ?

Reply:
Fixed to understand tabs